ls1intum/Artemis
# Broadcast team assignment updates via websockets

### Checklist

### Motivation and Context
At the moment the participation status is not being updated right away when students are assigned to teams. Even after a tutor has added them into a team, it will still say "You have not been assigned to a team yet" in the exercise card. The new team assignment status will only be reflected after the student refreshes the page. This PR adds broadcasting of team assignment changes via websockets so that students will see right away that they can start working on an exercise as soon as the tutor has created their team.

### Description
- Add `TeamWebsocketService` with method `sendTeamAssignmentUpdate` to broadcast changes in team assignments
- Use this service method when creating, updating, deleting or importing teams in `TeamResource`
- Add observable `teamAssignmentUpdates` to `team.service.ts` that emits team assignment updates
- Subscribe to this observable in the courses view and the course overview
- On receiving a team assignment update: update assigned team id, team participation and participation status on affected exercise

### Steps for Testing
Preconditions: A team-based exercise must exist in a course.

1. Log in as user A to Artemis (needs to be student in course)
2. Navigate to Course Overview
3. Click on a course that has a team-based exercise
4. In a different browser session, log in as user B (needs to be instructor in course)
5. Let user B assign and unassign user A from different teams for the team-based exercise
6. User A should see his participation status changing in the exercise row (from "You have not been assigned to a team yet" to "You have not started this exercise yet", or alternatively "No submission" etc. if a participation already exists, and vice-versa)

### Screencast
Behavior when adding and removing the user from the team:
![captured](https://user-images.githubusercontent.com/7109225/80873508-4dc7ce80-8cb9-11ea-893e-beeb1988955e.gif)
